drivers/builtin/src/chacha20.c

```diff
@@ -158,20 +157,22 @@ int mbedtls_chacha20_setkey(mbedtls_chacha20_context *ctx,
                             const unsigned char key[32])
 {
     /* ChaCha20 constants - the string "expand 32-byte k" */
-    ctx->state[0] = 0x61707865;
-    ctx->state[1] = 0x3320646e;
-    ctx->state[2] = 0x79622d32;
-    ctx->state[3] = 0x6b206574;
+    static const char EXPAND_32_BYTE_K[16] = "expand 32-byte k";
+    memcpy(ctx->state, EXPAND_32_BYTE_K, 16);
```

Comment:
I would think that on big endian machines this changes the values of the constants. What am I missing?

Reply:
Hmm. In `https://cr.yp.to/streamciphers/timings/estreambench/submissions/salsa20/chacha8/regs/chacha.c` this is written using:

```
static const char sigma[16] = "expand 32-byte k";
constants = sigma;
x->input[0] = U8TO32_LITTLE(constants + 0);
x->input[1] = U8TO32_LITTLE(constants + 4);
```

ie treat the string as a bunch of little-endian 32-bit ints. So it will write "e" into the lowest address, then "x", and so on, regardless of endianness. So unless I've missed something, this is equivalent behaviour to `memcpy`.

It does look like previously, the constant would have been written differently on big-endian, which looks like a bug.